bika/lims/utils/analysisrequest.py

```diff
@@ -116,18 +118,13 @@ def create_analysisrequest(client, request, values, analyses=None,
             )
         part_num += 1
 
-    # At this point, we have a fully created AR, with a Sample, Partitions and
-    # Analyses, but the state of all them is the initial ("sample_registered").
-    # We can now transition the whole thing (instead of doing it manually for
-    # each object we created). After and Before transitions will take care of
-    # cascading and promoting the transitions in all the objects "associated"
-    # to this Analysis Request.
-    sampling_workflow_enabled = sample.getSamplingWorkflowEnabled()
-    action = 'no_sampling_workflow'
-    if sampling_workflow_enabled:
-        action = 'sampling_workflow'
-    # Transition the Analysis Request and related objects to "sampled" (if
-    # sampling workflow not enabled) or to "to_be_sampled" statuses.
+    # At this point, we have AR, Sample, Partitions, Analyses,
+    # all in state "sample_registered". Manually execute initial
+    # transition for new objects:
+    auto = ar.bika_setup.getAutoReceiveSamples()
+    action = 'sampling_workflow' if sample.getSamplingWorkflowEnabled() \
+        else 'receive' if auto and isTransitionAllowed(ar, 'receive') \
+        else 'no_sampling_workflow'
```

Comment:
I agree this statement needs to be made easier to understand and read, I'll do.

Re using the guard, I feel the same problem as above; the guard should decide if the transition is allowed/available, not when the transition fires.  It will quickly get messy since "receive" guard/transition are available from multiple states. so the ugliness just moves.